File: packages/domain/lib/src/repositories/practice_session_repository.dart
import 'package:fmf_domain/src/models/practice_session.dart';

abstract interface class PracticeSessionRepository {
  Future<void> logSession(PracticeSession session);
  Future<List<PracticeSession>> getSessionsForSkill(String skillId);
  Future<List<PracticeSession>> getRecentSessions({int limit = 10});
}

File: apps/ios_app/Sources/Domain/Repositories/PracticeSessionRepository.swift
import Foundation

protocol PracticeSessionRepository: Sendable {
    func logSession(_ session: PracticeSession) async throws
    func deleteSession(id: String) async throws
    func getSessionsForSkill(_ skillId: String) async throws -> [PracticeSession]
    func getRecentSessions(limit: Int) async throws -> [PracticeSession]
}
